### ISSUE TITLE
'in' operator support not implemented for twisted.python.modules.PythonPath

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6198]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6198|
|Type|defect|
|Created|2012-11-30 17:09:39Z|
|Branch|https://github.com/twisted/twisted/tree/modules-contains-6198|

```
>>> from twisted.python.modules import theSystemPath
>>> theSystemPath["os"]
PythonModule<'os'>
>>> "os" in theSystemPath
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "twisted/python/modules.py", line 699, in __getitem__
    if '.' in modname:
TypeError: argument of type 'int' is not iterable
```

Attachments:

* [pythonpath-contains-6198.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/8e1/8e19d341c99f002bcb3cc09c06e3398420f29277/39ef90c20567cf2b4610d95de16a44678f9b9fc0.patch) (1424 bytes) - added by ephess on 2012-12-22 21:09:52Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6198 6198
type__defect defect
reporter__itamar itamar
priority__low low
milestone__ 
branch__branches_modules_contains_6198 branches/modules-contains-6198
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1354295379000000 1354295379000000
changetime__1356373895000000 1356373895000000
version__None None
owner__exarkun exarkun
cc__ephess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [36617]) Branching to 'modules-contains-6198'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [36618]) apply pythonpath-contains-6198.patch

refs [#6198](https://github.com/twisted/twisted/issues/6198)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

Thanks!  This looks pretty good.  I have a few comments though:

  1. A test should cover one case.  Since `unittest` uses exceptions to signal failures, any assertions that follow a failing assertion won't run.  This means failures may exist that won't get reported, if they happen to be checked after another failure.  Testing each case in a dedicated test method provides better reporting by showing all failures instead of just some (which means you don't have to run the test suite as many times, and you can be more confident that once you fix some failures, all tests will be passing).
  1. Use `L{}` to refer to functions, classes, or modules in a docstring.
  1. Test method docstrings should not use the word _should_.
  1. This new feature should be described by a news fragment file.

These are pretty straightforward changes.  I'll make them and, if buildbot is green, merge.  Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [36619]) address review comments

refs [#6198](https://github.com/twisted/twisted/issues/6198)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/modules-contains-6198)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [36622]) minor doc improvement

refs [#6198](https://github.com/twisted/twisted/issues/6198)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [36623]) Merge modules-contains-6198

Author: ephess
Reviewer: exarkun
Fixes: [#6198](https://github.com/twisted/twisted/issues/6198)

Implement the `in` operator for right-hand values of type `twisted.python.modules.PythonPath`.
